src/Bicep.Core/TypeSystem/StringLiteralType.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using Bicep.Core.Parser;
+
+namespace Bicep.Core.TypeSystem
+{
+    public class StringLiteralType : TypeSymbol
+    {
+        public StringLiteralType(string value)
+            : base(StringUtils.EscapeBicepString(value))
+        {
+            // The name of the type should be set to the escaped string value (including quotes).
+            // This affects how the type is displayed to the user, and is also used to compare two string literals types for equality.
```

Comment:
>compare two string literals [](start = 87, length = 27)

This is going to get weird when we implement unicode because there will be multiple bicep representations of the same string semantic value :(. On the other hand, TypeScript does show the escaped TS string with literal types.

I am wondering if we should internally store the .net value of the string internally (and use it for comparisons via the .net string comparison functions, but only expose the escaped version of it outside.

Reply:
That's true, and something we'll need to be careful about. In this implementation, I'm escaping from .NET representation to Bicep in order to generate the type string, so as long as our escape function is deterministic (it had better be!), this should always provide consistent comparisons even if the input strings have been specified using different encodings.

Reply:
Yeah